var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -123,17 +125,37 @@ def cmake_args(self):
 
         options = []
 
-        if self.spec.satisfies('@2020:'):
-            options.append('-DGMX_INSTALL_LEGACY_API=ON')
-
         if '+mpi' in self.spec:
             options.append('-DGMX_MPI:BOOL=ON')
-            # Ensures gmxapi builds properly
+            if self.version < Version('2020'):
+                # Ensures gmxapi builds properly
+                options.extend([
+                    '-DCMAKE_C_COMPILER=%s' % self.spec['mpi'].mpicc,
+                    '-DCMAKE_CXX_COMPILER=%s' % self.spec['mpi'].mpicxx,
+                    '-DCMAKE_Fortran_COMPILER=%s' % self.spec['mpi'].mpifc,
+                ])
+            elif self.version == Version('2021'):
```

Comment:
Note that these CMake options are contrary to the GROMACS installation guide, but are a necessary workaround for the 2021.0 release. Presumably, https://gitlab.com/gromacs/gromacs/-/merge_requests/1105 will be merged for 2021.1, which should allow the documented compiler hinting to work correctly.